### PR TITLE
Various fixes

### PR DESCRIPTION
**Pull request by AhmedSoliman**
- Detection of CatchingUp->Active fix.
- Retries for writing AnnounceLeader message
- Reducing logging verbosity in places
- Adding logs in others

### CODE REVIEW

File: crates/core/src/network/networking.rs [hunk 1]
```diff
@@ -181,16 +183,15 @@ impl<T: TransportConnect> NetworkSender<NoConnection> for Networking<T> {
...
-                        );
+                        if next_attempt >= max_attempts {
+                            trace!(
+                                "Connection to node {} failed with {}, next retry is attempt {}/{}",
+                                msg.peer(),
+                                e,
+                                next_attempt,
+                                max_attempts,
+                            );
+                        }
```
Comment (tillrohrmann) [inline crates/core/src/network/networking.rs:194]: Should this only be logged on the final attempt? If yes, then I guess we don't need to log `next retry is attempt {}/{}`.

File: crates/worker/src/partition/mod.rs [hunk 2]
```diff
@@ -531,7 +531,12 @@ where
         self.status.last_applied_log_lsn = Some(lsn);
         self.status.last_record_applied_at = Some(MillisSinceEpoch::now());
         match self.status.replay_status {
-            ReplayStatus::CatchingUp if self.status.target_tail_lsn.is_some_and(|v| lsn >= v) => {
+            ReplayStatus::CatchingUp
+                if self
+                    .status
+                    .target_tail_lsn
+                    .is_some_and(|tail| lsn.next() >= tail) =>
```
Comment (tillrohrmann) [inline crates/worker/src/partition/mod.rs:538]: Why `lsn.next()`? Wouldn't this set us to `Active` when we are processing `tail.prev()`?

**Review by tillrohrmann — APPROVED** [review]
The fixes look good to me. I had two questions where the change was not fully clear to me.

File: crates/worker/src/partition/mod.rs (hunk 2 above)
Comment (AhmedSoliman) [reply crates/worker/src/partition/mod.rs:538]: Yes. the last applied will be the point _before_ the tail

Comment (tillrohrmann) [reply crates/worker/src/partition/mod.rs:538]: At this point, we haven't fully processed this very last record, though. But I guess this solves the problem if there weren't any activity on the log after the `target_tail_lsn`, then we wouldn't move to active, right?

Comment (AhmedSoliman) [reply crates/worker/src/partition/mod.rs:538]: Exactly yes.

File: crates/core/src/network/networking.rs (hunk 1 above)
Comment (AhmedSoliman) [reply crates/core/src/network/networking.rs:194]: Right, let me remove this.

**Pull request merged**